airbyte-integrations/connectors/source-google-search-console/source_google_search_console/streams.py

```diff
@@ -123,7 +124,31 @@ def stream_slices(
 
         for site_url in self._site_urls:
             for search_type in self.search_types:
-                yield {"site_url": site_url, "search_type": search_type}
+                start_date = self._get_start_date(stream_state, site_url, search_type)
+                end_date = self._get_end_date()
+
+                if start_date > end_date:
+                    yield from [
+                        {
+                            "site_url": site_url,
+                            "search_type": search_type,
+                            "start_date": end_date.to_date_string(),
+                            "end_date": end_date.to_date_string(),
+                        }
+                    ]
+
+                next_start = start_date
+                period = pendulum.Duration(days=self.range_of_days)
+                while next_start <= end_date:
+                    next_end = min(next_start + period, end_date)
+                    yield {
+                        "site_url": site_url,
+                        "search_type": search_type,
+                        "start_date": next_start.to_date_string(),
+                        "end_date": next_end.to_date_string(),
+                    }
+                    # add 1 day for the next slice's start date not to duplicate data from previous slice's end date.
+                    next_start = next_end + pendulum.Duration(days=1)
```

Comment:
Why you add 1 day instead of period?

Reply:
The period is added [here](https://github.com/airbytehq/airbyte/pull/9073/files#diff-f69df367010e34672c57c224277cb6731109c1869ca1ba980a6f4c1067dda837R143)

Without this line
`next_start = next_end + pendulum.Duration(days=1)`

The two slices will intersect, then user gets duplicated records:
{"start_date": "2021-09-01", "end_date": "_2021-09-02"_}
{"start_date": "_2021-09-02_", "end_date": "2021-09-03"}